### PR TITLE
Save single component and grader/grade time on numeric/checkpoint grading

### PR DESCRIPTION
**Pull request by MasterOdin**
This closes the first two points of #1004 (though I'm not displaying this information anywhere yet, just saving it to the DB) as well as moving towards a better model system (where the models themselves are responsible for calling the appropriate DB methods to store their information (or not) instead of the controllers). The GradeableComponent itself now contains two new fields `\app\models\User $grader` and `\DateTime grade_time` that contains this information.

This also makes it so that clicking a checkpoint or editing a numeric/text will only submit an ajax call to update that one column (instead of the whole row) so as to support this feature.

Additionally added a new method on `__call()` as well, `is*` (ex: `isModified()`) which will always return the result of `* === true` (useful for boolean flags on a model). Will probably in a later commit reduce down the `is_extra_credit` and `is_text` flags in GradeableComponent to just be `extra_credit` and then reference it via the `isExtraCredit()` magic function.

### CODE REVIEW

**Comment by jbarthelmess** [comment]
closes #1041 
makes progress on #856

**Comment by MasterOdin** [comment]
This actually probably breaks submitting an electronic gradeable rubric as I removed a column that it's saving. Need to fix that before this is accepted.

File: site/app/libraries/database/DatabaseQueriesPostgresql.php [hunk 1]
```diff
@@ -814,37 +835,34 @@ public function updateActiveVersion($g_id, $user_id, $team_id, $version) {
...
+        $params = array($gradeable->getId(), $gradeable->getUser()->getId(),
+                        $gradeable->getOverallComment(), $gradeable->getStatus(), 0,
+                        $gradeable->getActiveVersion());
+        $this->database->query("INSERT INTO 
+gradeable_data (g_id, gd_user_id, gd_overall_comment, gd_status, gd_late_days_used, gd_active_version)
+VALUES (?, ?, ?, ?, ?, ?)", $params);
+        return $this->database->getLastInsertId("gradeable_data_gd_id_seq");
+    }
+
     public function updateGradeableData(Gradeable $gradeable) {
```
Comment (jbarthelmess) [inline site/app/libraries/database/DatabaseQueriesPostgresql.php]: Why leave this empty function in?

Comment (MasterOdin) [inline site/app/libraries/database/DatabaseQueriesPostgresql.php]: Discussed this today, I'm leaving it in as it'll eventually be used (when we are submitting electronic gradeable rubrics where the gradeable_data table will actually warrant a change and this function can be properly tested), just not at this exact moment.

**Comment by MasterOdin** [comment]
I weakened the gd_grader_id column to be nullable (since it's also going away) which should fix the error you were seeing.

**Pull request merged**